src/Transform/ONNX/Decompose.td

```diff
@@ -24,159 +24,177 @@ include "src/Dialect/ONNX/ONNXOps.td"
+
+def ScalerT : NativeCodeCall<"TypeAttr::get($_builder.getF32Type())">;
+
+def noop_with_empty_axes
+    : NativeCodeCall<"IntegerAttr::get($_builder.getIntegerType(64, "
+                     "/*isSigned=*/true), APInt(64, 0, /*isSigned=*/true))">;
 
 //===----------------------------------------------------------------------===//
 // ONNXReduceL1Op %X = ONNXReduceSumOp (ONNXAbsOp %X)
 //===----------------------------------------------------------------------===//
-def ReduceL1OpPattern: Pat<(ONNXReduceL1Op $oprd, $axes, $keepdims),
-                           (ONNXReduceSumOp (ONNXAbsOp $oprd), $axes, $keepdims)>;
+def ReduceL1OpPattern
+    : Pat<(ONNXReduceL1Op $oprd, $axes, $keepdims),
+          (ONNXReduceSumOp(ONNXAbsOp $oprd),
+              (ONNXConstantOpFromDenseAttr(createDenseArrayAttrOrEmpty $axes)),
+              $keepdims, (noop_with_empty_axes))>;
 
 //===----------------------------------------------------------------------===//
 // ONNXReduceL2Op %X = ONNXSqrtOp (ONNXReduceSumSquareOp (%X))
 //===----------------------------------------------------------------------===//
-def ReduceL2OpPattern: Pat<(ONNXReduceL2Op $oprd, $axes, $keepdims),
-                           (ONNXSqrtOp (ONNXReduceSumSquareOp $oprd, $axes, $keepdims))>;
+def ReduceL2OpPattern
+    : Pat<(ONNXReduceL2Op $oprd, $axes, $keepdims),
+          (ONNXSqrtOp(ONNXReduceSumSquareOp $oprd, $axes, $keepdims))>;
 
 //===----------------------------------------------------------------------===//
 // ONNXReduceLogSumOp %X = ONNXLogOp (ONNXReduceSumOp (%X))
 //===----------------------------------------------------------------------===//
-def ReduceLogSumOpPattern: Pat<(ONNXReduceLogSumOp $oprd, $axes, $keepdims),
-                           (ONNXLogOp (ONNXReduceSumOp $oprd, $axes, $keepdims))>;
+def ReduceLogSumOpPattern
+    : Pat<(ONNXReduceLogSumOp $oprd, $axes, $keepdims),
+          (ONNXLogOp(ONNXReduceSumOp $oprd,
+              (ONNXConstantOpFromDenseAttr(createDenseArrayAttrOrEmpty $axes)),
+              $keepdims, (noop_with_empty_axes)))>;
 
 //===----------------------------------------------------------------------===//
 // ONNXReduceLogSumExpOp %X
 //===----------------------------------------------------------------------===//
 // keepdims is true
-def ReduceLogSumExpOpPattern1: Pat<
-  (ONNXReduceLogSumExpOp $oprd, $axes, $keepdims),
-  (ONNXAddOp
-    (ONNXReduceLogSumOp
-      (ONNXExpOp
-         (ONNXSubOp $oprd, (ONNXReduceMaxOp:$max $oprd, $axes, (KeepdimsTrue)))),
-      $axes,
-      $keepdims),
-    $max),
-  [(KeepdimsIsTrue:$keepdims)],
-  (addBenefit 1)
->;
-
-// keepdims is false 
-def ReduceLogSumExpOpPattern2: Pat<
-  (ONNXReduceLogSumExpOp $oprd, $axes, $keepdims),
-  (ONNXAddOp
-    (ONNXReduceLogSumOp
-      (ONNXExpOp
-         (ONNXSubOp $oprd, (ONNXReduceMaxOp:$max $oprd, $axes, (KeepdimsTrue)))),
-      $axes,
-      $keepdims),
-    (ONNXSqueezeOp $max, $axes)),
-  [],
-  (addBenefit 0)
->;
+def ReduceLogSumExpOpPattern1
```

Comment:
Did you add a lit test for these new patterns?

Reply:
Not new patterns. Just replaced the old ReduceSum with the new version. Backend test contains cases for this pattern.